Add e2e tests for storageclass

jsafrane opened the pull request:
- test pd-ssd and pd-standard on GCE,
- test all four volume types and encryption on AWS
- test just the default volume type on OpenStack (right now, there is no API
  to get list of them)

These tests are quite slow, e.g. there are two tests on AWS that has to run mkfs.ext4 on 500 GB magnetic drive with low IOPS, which takes ~3-4 minutes each.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32485)

<!-- Reviewable:end -->

jsafrane commented:
cc @kubernetes/sig-storage

gnufied commented on test/e2e/volume_provisioning.go:
```diff
@@ -114,45 +156,170 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 	framework.KubeDescribe("DynamicProvisioner Alpha", func() {
-		It("should create and delete alpha persistent volumes [Slow]", func() {
+		It("should provision alpha volumes [Slow]", func() {
 			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
```
So alpha here just means provisioning without storageclass right?

jsafrane replied:
yes

gnufied commented on test/e2e/volume_provisioning.go:
```diff
@@ -114,45 +156,170 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
+				"500Gi", // minimum for sc1
+				"500Gi",
+				func(volume *api.PersistentVolume) error {
+					return checkAWSEBS(volume, "sc1", false)
```
Do you think it may be worth replacing this with `nil` like openstack checks because it doesn't do anything?

gnufied commented on test/e2e/volume_provisioning.go:
```diff
@@ -101,6 +128,21 @@ func testDynamicProvisioning(client *client.Client, claim *api.PersistentVolumeC
 }
 
+func checkAWSEBS(volume *api.PersistentVolume, volumeType string, encrypted bool) error {
+	return nil
```
Intended?

jsafrane replied:
I forgot to implement AWS checks. I just added them to the PR.

gnufied commented on pkg/cloudprovider/providers/aws/aws.go:
```diff
@@ -1606,6 +1606,34 @@ func (c *Cloud) DetachDisk(diskName string, nodeName types.NodeName) (string, er
+		return fmt.Errorf("expected encrypted volume, got %v", *info.Encrypted)
+	}
+	return nil
+}
```
Minor nit - since this will be strictly used from tests - do you think we can just move this to a test helper or something?  But I am partly okay with this being here too.. so not a deal breaker. :-)

gnufied reviewed (commented):
LGTM with minor comments.

jsafrane committed:
Add e2e tests for storageclass

- test pd-ssd and pd-standard on GCE,
- test all four volume types on AWS
- test just the default volume type on OpenStack (right now, there is no API
  to get list of them)

childsb commented:
This looks good.

zmerlynn commented:
This broke hard AWS as well. I'm going to send in a revert, feel free to race me to fix it. https://k8s-testgrid.appspot.com/google-aws
